lib/blocks.php

```diff
@@ -93,3 +93,104 @@ function do_blocks( $content ) {
 	return $content_after_blocks;
 }
 add_filter( 'the_content', 'do_blocks', 9 ); // BEFORE do_shortcode() and wpautop().
+
+/**
+ * Extract the blocks from post content for the REST API post response.
+ *
+ * @since 1.1.0
+ *
+ * @param string $content The post content.
+ *
+ * @return array Array of block data.
+ */
+function gutenberg_add_blocks_to_post_resource( $content ) {
+	$registry = WP_Block_Type_Registry::get_instance();
+	$blocks   = gutenberg_parse_blocks( $content );
+	$data     = array();
+
+	// Loop thru the blocks, adding rendered content when available.
+	foreach ( $blocks as $block ) {
+		$block_name  = isset( $block['blockName'] ) ? $block['blockName'] : null;
+		$attributes  = is_array( $block['attrs'] ) ? $block['attrs'] : null;
+		$raw_content = isset( $block['rawContent'] ) ? $block['rawContent'] : null;
+
+		// Skip block if we didn’t get a valid block name.
+		if ( null === $block_name ) {
+			continue;
+		}
+
+		// Set up rendered content, if available.
+		$block['renderedContent'] = null;
+		$block_type = $registry->get_registered( $block_name );
+		if ( null !== $block_type ) {
+			$block['renderedContent'] = $block_type->render( $attributes, $raw_content );
+		}
+
+		// Add the item data.
+		$data[] = array(
+			'type'       => $block_name,
+			'attributes' => $attributes,
```

Comment:
I'm not able to view this myself due to previously noted issues with the branch but... what are we expecting `attributes` to be here? The server is only capable of parsing attributes serialized into the comment demarcations, which is an incomplete representation.

Reply:
I'm not certain what these are expected to be - these are returned from `gutenberg_parse_blocks` as `attrs` and I am transforming the name here to be consistent with #2503.